src/lib.rs

```diff
@@ -0,0 +1,1012 @@
+		self
+	}
+
+	/// Sets the Bitcoin network used.
+	///
+	/// Options: `mainnet`/`bitcoin`, `testnet`, `regtest`, `signet`
+	///
+	/// Default: `testnet`
+	pub fn set_network(&mut self, network: &str) -> &mut Self {
+		self.config.network = match network {
+			"mainnet" => bitcoin::Network::Bitcoin,
+			"bitcoin" => bitcoin::Network::Bitcoin,
+			"testnet" => bitcoin::Network::Testnet,
+			"regtest" => bitcoin::Network::Regtest,
+			"signet" => bitcoin::Network::Signet,
+			_ => bitcoin::Network::Testnet,
+		};
+		self
+	}
+
+	/// Sets the port on which [`LdkLite`] will listen for incoming network connections.
+	///
+	/// Default: `9735`
+	pub fn set_listening_port(&mut self, listening_port: u16) -> &mut Self {
+		self.config.listening_port = listening_port;
+		self
+	}
+
+	/// Builds an [`LdkLite`] instance according to the options previously configured.
+	pub fn build(&self) -> LdkLite {
+		let config = Arc::new(self.config.clone());
+
+		let ldk_data_dir = format!("{}/ldk", &config.storage_dir_path.clone());
+		fs::create_dir_all(ldk_data_dir.clone()).expect("Failed to create LDK data directory");
+
+		let bdk_data_dir = format!("{}/bdk", config.storage_dir_path.clone());
+		fs::create_dir_all(bdk_data_dir.clone()).expect("Failed to create BDK data directory");
+
+		// Step 0: Initialize the Logger
+		let log_file_path = format!("{}/ldk_lite.log", config.storage_dir_path.clone());
+		let logger = Arc::new(FilesystemLogger::new(log_file_path));
+
+		// Step 1: Initialize the on-chain wallet and chain access
+		let seed = io_utils::read_or_generate_seed_file(Arc::clone(&config));
+		let xprv = bitcoin::util::bip32::ExtendedPrivKey::new_master(config.network, &seed)
+			.expect("Failed to read wallet master key");
+
+		let wallet_name = bdk::wallet::wallet_name_from_descriptor(
+			Bip84(xprv.clone(), bdk::KeychainKind::External),
+			Some(Bip84(xprv.clone(), bdk::KeychainKind::Internal)),
+			config.network,
+			&Secp256k1::new(),
+		)
+		.expect("Failed to derive on-chain wallet name");
+		let database = sled::open(bdk_data_dir).expect("Failed to open BDK database");
+		let database =
+			database.open_tree(wallet_name.clone()).expect("Failed to open BDK database");
+
+		let bdk_wallet = bdk::Wallet::new(
+			Bip84(xprv.clone(), bdk::KeychainKind::External),
+			Some(Bip84(xprv.clone(), bdk::KeychainKind::Internal)),
+			config.network,
+			database,
+		)
+		.expect("Failed to setup on-chain wallet");
+
+		// TODO: Check that we can be sure that the Esplora client re-connects in case of failure
+		// and and exits cleanly on drop. Otherwise we need to handle this/move it to the runtime?
+		let blockchain = EsploraBlockchain::new(&config.esplora_server_url, BDK_CLIENT_STOP_GAP)
+			.with_concurrency(BDK_CLIENT_CONCURRENCY);
+
+		let chain_access = Arc::new(ChainAccess::new(blockchain, bdk_wallet, Arc::clone(&logger)));
+
+		// Step 3: Initialize Persist
+		let persister = Arc::new(FilesystemPersister::new(ldk_data_dir.clone()));
```

Comment:
Right, simply because I haven't come around to wrapping the `sled` database in a reasonable abstraction and implementing the necessary trait on it. I'm kind of holding off on this a bit since a) BDK seems to suggest using SQLite now and b) BDK's current use of the database makes `bdk::Wallet` not `Send`, which is messing with `async`. This is however due to change in BDK core / 1.0, which we therefore should adopt ASAP. 

So, I so far was holding off touching the database things too much to avoid finding a workaround (and possibly introducing some trade-offs), just to have to change it again anyways. But, will definitely change this in an upcoming PR soon.